### PR TITLE
feat(styles)!: L3-2994 update text styles and link styles for new About Us Figma

### PR DESCRIPTION
**Pull request by scottdickerson**
BREAKING CHANGE: TextVariants and their tokens were changed to remove Display* tokens and replace them with Title* tokens.   LinkVariant options were removed, most importantly `navLinkSm` and `navMenu`, and `navLinkLg` replaced with the `snwFlyoutLink` and `snwHeaderLink`.  `Standalone` and `inline` were replaced with `link`

**Jira ticket**

[L3-2994](https://phillipsauctions.atlassian.net/browse/L3-2994)

**Screenshots**
Please ignore the header styles issues as they will be fixed in a follow-on PR.

| Before | After |
| ------ | ----- |
| image | image |

**Summary**
The About Us [Figma ](https://www.figma.com/design/Hp2FyltbOmRxTuw9kSwBAd/EPIC-About-Us?node-id=1265-4115&m=dev), Text, Hero banner, and Links format has changed since the original design.  

This also updates the underlying seldon tokens to removed any tokens no longer listed  [here](https://www.figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?node-id=7263-1361&m=dev) in the new token design so consumers don't try and use them and all the SCSS referencing them is removed

**Change List (describe the changes made to the files)**

- change(--display*): Totally remove the `display` tokens and variants as they have been replaced by the `title`variants.
- change(LinkVariants): pare down any deprecated link variants.
- change(HeroBanner): reuse existing Text components and add the linear gradient as part of the component itself.  Also fix it to look like the Mobile breakpoint version shown in our Component Library
- change(Link): update to match the reduced list of variants
- change(type-tokens): Our heading tokens have been changed and a new `heading5` has been created, update the documentation here

**Acceptance Test (how to verify the PR)**

- Verify the Text stories, and the Link, and the NavLink stories, make sure they match the Figma in AboutUs
- Check NavigationList and NavItem stories (bounce will be fixed in the Header styles update PR)
- (Header stories still need work in a separate PR)
- Inspect the CSS in dev tools and search for  `display0` in the generated CSS

**Regression Test**

- N/A

**Evidence of testing**

- I did a local `npm link @phillips/seldon` so that I could test in the About Us page and compare it vs the Figma in the screenshots above.

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-2994]: https://phillipsauctions.atlassian.net/browse/L3-2994?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by constantinehuzenko** [comment]
we have small bug on `snwFlyoutLink`

image

File: src/components/Navigation/NavigationItem/NavigationItem.tsx [hunk 1]
```diff
@@ -56,7 +56,11 @@ const NavigationItem = ({
       data-testid={`nav-item-${label}`}
       className={classNames(`${px}-nav__item`, navGroup, className)}
     >
-      <Component href={href} variant={navType ? navType : LinkVariants.navMain} tabIndex={expandedItem === '' ? 0 : -1}>
+      <Component
+        href={href}
+        variant={navType ? navType : LinkVariants.snwHeaderLink}
+        tabIndex={expandedItem === '' ? 0 : -1}
```
Comment (constantinehuzenko) [inline src/components/Navigation/NavigationItem/NavigationItem.tsx:62]: can we move all ternary to variables?
```
const variant = navType ? navType : LinkVariants.navMain
const tabIndex = expandedItem === '' ? 0 : -1
const variant = navType ? navType : LinkVariants.snwHeaderLink
const tabIndex = expandedItem === '' ? 0 : -1
```

**Review by constantinehuzenko — COMMENTED** [review]
thank you for removing unnecessary parts
just left couple minor comments

File: src/components/Link/utils.ts [hunk 2]
```diff
@@ -1,27 +1,9 @@
...
 
-export const getLinkVariantClassName = (variant: keyof typeof UpdatedLinkVariants) => `${px}-link--${variant}`;
+export const getLinkVariantClassName = (variant: keyof typeof LinkVariants) => `${px}-link--${variant}`;
 
 export const isLinkExternal = (href?: string) =>
   !!href?.match(
     /(http[s]?:\/\/)(?!.*phillips\.com)([a-zA-Z0-9\-.]+)(:[0-9]{1,4})?([a-zA-Z0-9/\-._~:?#[\]@!$&'()*+,;=]*)/g,
   );
-
-// Leaving here for backwards compatibility. Can remove on next major release
```
Comment (davidicus) [inline src/components/Link/utils.ts:11]: breaking semver here. Should we merge this and the Prefix change together and release v2?

File: src/components/Text/utils.ts [hunk 3]
```diff
@@ -3,17 +3,15 @@ import { TextVariants } from './types';
...
 };
 
 export const determineDefaultTextElement = (variant: TextVariants = TextVariants.body1): string => {
-  if ([TextVariants.display0, TextVariants.display1].includes(variant)) return 'h1';
-  if ([TextVariants.display2, TextVariants.display3, TextVariants.display4].includes(variant)) return 'h3';
   if (variant === TextVariants.blockquote) return 'blockquote';
-
   if (variant.includes('body')) return 'p';
+  if (variant.includes('string')) return 'span';
+  if (variant.includes('title')) return 'span';
```
Comment (davidicus) [inline src/components/Text/utils.ts:14]: seems like a strange choice for title? Is this on purpose?

**Review by davidicus — CHANGES_REQUESTED** [review]
Some minor comments. Plus, none of our fonts are loading anymore. Looking at the [link variant story ](https://deploy-preview-282--phillips-seldon.netlify.app/?path=/story/components-links-link--playground)you can see they are all broken and falling back to the default and assigned fallback.

**Comment by scottdickerson** [comment]
> we have small bug on `snwFlyoutLink`
> 
> image

What's the small bug @constantinehuzenko ?

File: src/components/Link/utils.ts (hunk 2 above)
Comment (scottdickerson) [reply src/components/Link/utils.ts:11]: each one will make it's own version, but I will rev this one.

File: src/components/Navigation/NavigationItem/NavigationItem.tsx (hunk 1 above)
Comment (scottdickerson) [reply src/components/Navigation/NavigationItem/NavigationItem.tsx:62]: in this case I'd rather do them inline

**Comment by scottdickerson** [comment]
> Some minor comments. Plus, none of our fonts are loading anymore. Looking at the [link variant story ](https://deploy-preview-282--phillips-seldon.netlify.app/?path=/story/components-links-link--playground)you can see they are all broken and falling back to the default and assigned fallback.

I'm not sure why or when this happened, but I've fixed it now.

File: src/components/HeroBanner/_heroBanner.scss [hunk 4]
```diff
@@ -2,22 +2,21 @@
 
 .#{$px}-hero-banner {
   align-items: center;
-  background: var(--background);
+  background: linear-gradient(rgba(0, 0, 0, 60%), rgba(0, 0, 0, 60%)), var(--background);
```
Comment (davidicus) [inline src/components/HeroBanner/_heroBanner.scss:5]: Are we sure we want to hard code the linear gradient? I know it sucks to write these but if we wanted to just provide a linear gradient as a background for whatever reason we will not be able to overwrite this. I do not think we have a current use case, just wondering if we want to keep the flexibility.

**Comment by constantinehuzenko** [comment]
> > we have small bug on `snwFlyoutLink`
> > image
> 
> What's the small bug @constantinehuzenko ?

It was problem with underline in snwFlyoutLink

File: src/components/Text/utils.ts (hunk 3 above)
Comment (scottdickerson) [reply src/components/Text/utils.ts:14]: @davidicus as we discussed, we're not sure where to use `title` yet

File: src/components/HeroBanner/_heroBanner.scss (hunk 4 above)
Comment (scottdickerson) [reply src/components/HeroBanner/_heroBanner.scss:5]: I'm assuming they'll want this consistency, but I think we adjust later if needed.

File: src/design/type-tokens/type-tokens.mdx [hunk 5]
```diff
@@ -20,216 +20,216 @@ consistent and scalable design system.
...
   <summary>Example</summary>
+</details>
 
 ```scss
-@include text($display0);
+@include text($heading1);
 ```
 
-  <span className="example example--d0">example</span>
-</details>
```
Comment (davidicus) [inline src/design/type-tokens/type-tokens.mdx:50]: I think this is the part that broke the example.

**Review by davidicus — CHANGES_REQUESTED** [review]
There are CSS issues with the header but I assume that will be fixed with the other tickets. The [first example ](https://deploy-preview-282--phillips-seldon.netlify.app/?path=/docs/design-type-tokens--overview#heading)in the type token documentation does not render the example. Other than the broken example I think this is good to go.

File: src/design/type-tokens/type-tokens.mdx (hunk 5 above)
Comment (scottdickerson) [reply src/design/type-tokens/type-tokens.mdx:50]: this is fixed now @davidicus

**Comment by scottdickerson** [comment]
I find this disabled state pretty hard to read: 
image

Could we make it a bit darker?

**Review by davidicus — APPROVED** [review]
My points have been addressed.

**Pull request merged**